app/addons/documents/index-editor/reducers.js

```diff
@@ -0,0 +1,196 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+import ActionTypes from './actiontypes';
+import Resources from "../resources";
+
+const defaultMap = 'function (doc) {\n  emit(doc._id, 1);\n}';
+const defaultReduce = 'function (keys, values, rereduce) {\n  if (rereduce) {\n    return sum(values);\n  } else {\n    return values.length;\n  }\n}';
+const builtInReducers = ['_sum', '_count', '_stats'];
```

Comment:
We have a new built in reduce `_approx_count_distinct` http://docs.couchdb.org/en/stable/ddocs/ddocs.html?highlight=reduce#built-in-reduce-functions

Reply:
I'll do this in a separate PR because it should only allow this option when the server version is >= 2.2.0.